libraries/DataFlash/LogFile.cpp

```diff
@@ -1702,6 +1709,45 @@ void DataFlash_Class::Log_Write_ESC(void)
         }
     }
 #endif // CONFIG_HAL_BOARD
+
+#if HAL_WITH_UAVCAN
+    for (uint8_t i = 0; i < MAX_NUMBER_OF_CAN_INTERFACES; i++) {
+        if (hal.can_mgr[i] == nullptr) {
+            continue;
+        }
+        AP_UAVCAN *ap_uavcan = hal.can_mgr[i]->get_UAVCAN();
+        if (ap_uavcan == nullptr) {
+            continue;
+        }
+        if (!ap_uavcan->escstatus_sem_take()) {
```

Comment:
I'm OK with how this is, but would it make more sense to do the escstatus_sem_take outside of the loop?

Reply:
Not possible at all.